providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/file/FileConnector.java

```diff
@@ -0,0 +1,58 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.file;
+
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.Connector;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayload;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayloadType;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * File connector reads flag configurations and expose the context through {@code Connector} contract.
+ * The implementation is kept minimal and suites testing, local development needs.
+ */
+@SuppressFBWarnings(value = {"EI_EXPOSE_REP", "PATH_TRAVERSAL_IN"},
+        justification = "File connector read feature flag from a file source.")
+@Slf4j
+public class FileConnector implements Connector {
```

Comment:
Good to see how well this abstraction worked!

Reply:
Yes and thank you for proposing this.

And, we can easily expand this to a "file watcher" mode.